ras_party/controllers/account_controller.py

```diff
@@ -746,6 +746,15 @@ def put_email_verification(token, tran, session):
         # We set the user as verified on the OAuth2 server.
         set_user_verified(email_address)
 
+    if not respondent.pending_email_address and respondent.password_verification_token:
```

Comment:
This feels in the wrong place and order. The issue materializes because the email in the token has the original email and due to the pending becoming the new email makes it makes it invalid. With that in mind it should be there and not here. Having it here puts in on the original verification path which isn't valid. 

It should be done inside update_verified_email_address or alternatively after line 727

Reply:
This has been moved up to line 729 right after the email update